### PR TITLE
Docker service logs options

### PR DESCRIPTION
**Pull request by Zinggi**
We are sometimes running into a timeout probably still caused by [this docker bug](https://github.com/moby/moby/issues/41733) when printing docker service logs.

This problem was already addressed with https://github.com/sudo-bmitch/docker-stack-wait/pull/23, which has helped, but makes our affected pipelines slow and fail, even if a deployment succeeds.  

It looks like the `--tail` option of `docker service logs` is just very buggy, so the only workaround is to avoid it completely.
But I still want to see log output at the end, so I want to use `--since` instead, which is not affected by this bug.

To make this work, I added a new option which just passes the flag directly to `docker service logs`, then we can just switch to using `docker-stack-wait -l '--since 2m'` in our pipelines to work around the issue.

I published my fork on dockerhub as `zinggi57/docker-stack-wait:docker-service-logs-options`: https://hub.docker.com/r/zinggi57/docker-stack-wait/tags and started using that with the `-l '--since 2m'` in our pipelines, which solves the timeout issue for us.

You can drop my last commit if you don't want to add a deprecated notice for the `-p` flag, but I thought it makes sense to remove that at some point because the new one is more powerful.

### CODE REVIEW

**Comment by Zinggi** [comment]
@sudo-bmitch sorry for the ping, but is there any chance you could take a look at this?

We've been using my branch at work for a while now and we haven't seen any timeout issues since.

**Review by sudo-bmitch — CHANGES_REQUESTED** [review]
See the requested changes. This can be squashed into a single commit.

**Comment by Zinggi** [comment]
@sudo-bmitch thanks for taking a look!
I didn't know about the sorting convention, should be all good now :+1:

**Review by sudo-bmitch — APPROVED** [review]
Thanks, LGTM!

**Comment by Zinggi** [comment]
Nice, thanks a lot!

**Comment by ryudoawaru** [comment]
Hi, this change makes my deployment continues to print logs and my GitLab CI treats it as never ending.
截圖 2023-10-12 下午9 33 04

How could I fix this problem?

**Comment by ryudoawaru** [comment]
Thanks for reply, I used to use just `-r $DEPLOY_NAME` only.

I am trying to add `-l 0 -r $DEPLOY_NAME` to avoid this issue, will let you know once I get the result.

File: docker-stack-wait.sh [hunk 1]
```diff
@@ -78,20 +79,21 @@ service_state() {
   fi
 }
 print_service_logs() {
-  if [ "$opt_p" != "0" ]; then
+  if [ "$opt_l" != "0" ]; then
```
Comment (Zinggi) [inline docker-stack-wait.sh:82]: Ahh, I see the mistake now.
Yeah, looks like I accidentally changed the default options, so it now does `docker service logs  "$service_id"`, which never stops.

Sorry about that!

Comment (ryudoawaru) [reply docker-stack-wait.sh:82]: Never mind, I am glad to be able to help this project.

**Comment by Zinggi** [comment]
This should fix it: https://github.com/sudo-bmitch/docker-stack-wait/pull/26

**Pull request merged**